pkg/stub/handler.go

```diff
@@ -25,12 +29,15 @@ type Handler struct {
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
-	if event.Deleted {
-		return nil
-	}
 	switch o := event.Object.(type) {
 	case *ingressv1alpha1.ClusterIngress:
-		return h.syncIngressUpdate(o)
+		if event.Deleted {
```

Comment:
Prefer using switch and case that will give clear picture of which events are handled or ignored (general pattern used in openshift for handling events)
```
switch event.Type {
case event.Added, event.Modified: (or event.Synced, check what is exposed by sdk.Event)
...
case event.Deleted:
}
```

Reply:
sdk.Event is different from watch.Event, it only sets Deleted flag in case of deletion and Object is the latest in case of add or modify event (no other Added, Modified or Synced events).
Ignore my comments, your implementation looks good.